Makefile

```diff
@@ -12,4 +13,18 @@ build:
 clean:
 	rm -rf proton_build proton_install qpid-dispatch.tar.gz qpid-dispatch-src qpid-proton.tar.gz qpid-proton-src staging build
 
-.PHONY: build clean
+cleanimage:
+	docker image rm -f qdrouterd-builder
+
+push:
+ifeq ($(strip $(DOCKER_USER)),)
+$(error DOCKER_USER not set)
+endif
+ifeq ($(strip $(DOCKER_PASSWORD)),)
+$(error DOCKER_PASSWORD not set)
+endif
+	docker build -t quay.io/interconnectedcloud/qdrouterd:latest .
+	@docker login -u ${DOCKER_USER} -p ${DOCKER_PASSWORD} ${DOCKER_REGISTRY}
```

Comment:
I think if you have already logged in once, that the credentials are stored against the registry and don't need to be entered again. If so could possibly leave the loging step out. Unless of course the intention is to use this from CI... perhaps it could be optional? I.e. login if username and password provided, otherwise assume it is not required?

Reply:
Make sense. I pushed up a new commit to this PR which uses the environment variables if available otherwise assumes that the user is already logged in.